### ISSUE TITLE
Easiest way to "step" in script execution?

### ISSUE THREAD

**Issue by elliot-nelson** [engine262/engine262#149]
Hi there, love the project, have been experimenting with it locally and it's really compelling.

What would be the simplest way to be able to create a realm, load up a script, and then repeatedly "step" through each expression completion?  I've looked at `NormalCompletion` and it seems like I could potentially override it, but I'm not sure how to call a user-created callback from here.  (Or, how to see what line/column # in the original source I'm at at that completion step).

A potentially different approach might be to somehow insert `debugger` in between every expression and define an `onDebugger` call.

TLDR: If it's not already supported, what path would you take to using this engine to make a debugger-style "step through code" feature?

**Comment by elliot-nelson** [comment]
Ah I think I posted a minute too soon... if I'm reading it right, this test is a template: https://github.com/engine262/engine262/blob/main/test/stepped.js

I'll play with it and see if this works for my use case!

**Comment by devsnek** [comment]
Seems this was figured out
